chroot_setup.py

```diff
@@ -0,0 +1,373 @@
+  ':mknod'      : chroot_mknod,      # done
+  ':cp_dir'     : chroot_cp_dir,     # done
+  ':cp'         : chroot_cp,         # done
+  ':chmod_dir'  : chroot_chmod_dir,  # done
+  ':chmod'      : chroot_chmod,      # done
+  ':chown_dir'  : chroot_chown_dir,  # done
+  ':chown'      : chroot_chown,      # done
+  ':system'     : chroot_system,
+  ':sh'         : chroot_sh
+}
+
+def chroot_setup (chroot_lst):
+  lst_file = open(chroot_lst, 'r')
+  action = None
+
+  # Iterate over all lines in the file, executing actions where encountered
+  for line in lst_file:
+    line = line.strip()
+
+    # Blank line
+    if not line:
+      # print('empty')
+
+    # Comment
+    else if line.startswith('#'):
+      # print('comment')
+
+    # Determine action
+    else if line.startswith(':'):
+      action = action_dictionary[line]
+
+    # Execute action
+    else:
+      if action not None:
+        action(line)
+
+# TODO: documentation
+def initial_check ():
+  if (os.path.exists('/QOpenSys/usr/bin')):
+    CHROOT_DEBUG = False
+    PATH='/QOpenSys/usr/bin:/QOpenSys/usr/sbin:'
+    LIBPATH='/QOpenSys/usr/lib'
+    os.environ['PATH'] = PATH
+    os.environ['LIBPATH'] = LIBPATH
+    print("**********************")
+    print("Live IBM i session (changes made).")
+    print("**********************")
+
+    if (ags.verbose):
+      print('PATH=' + PATH)
+      print('LIBPATH=' + LIBPATH)
+
+  else:
+    CHROOT_DEBUG = True
+    print("**********************")
+    print("Not IBM i, no action is taken (debug flow purpose only).")
+    print("**********************")
+    # exit -1
+
+
+def yum_install(package_list: list):
+  os.environ["LIBPATH"] = '/QOpenSys/pkgs/lib:/QOpenSys/usr/lib'
+  install_template = Template('/QOpenSys/pkgs/bin/yum $yes --installroot=$chroot_dir install $package')
+  auto_yes = '-y' if args.auto_yes else ''
+  for package in package_list:
+    command = install_template.substitute(yes=auto_yes, chroot_dir=args.chroot_directory, package=package)
+    logger.debug(command)
+    os.system(command)
+
+#
+# MAIN
+#
+
+lst = 0
+ops = 0
+qsys = 0
+CHROOT_DIR = ""
+CHROOT_LIST = ""
+SCRIPT_DIR = os.path.dirname(os.readlink(args.chroot_directory)) + "/config"
+
+# Didn't implement all of the checks, argparse can do that
+
+intial_check()
+
+# This is done because the arg may be a relative path name.
+# Need to be done before any 'cd'
+TEMP_DIR = os.readlink(args.chroot_directory)
+
+logger.debug("Scipt Dir: " + SCRIPT_DIR)
+os.chdir(SCRIPT_DIR)
+logger.debug("PWD: " + os.getcwd())
+
+if not TEMP_DIR.startswith('/QOpenSys/'):
+  print("\nERROR: [CHROOT DIRECTORY] Must begin with /QOpenSys/...\n")
+  parser.print_help
+  sys.exit(-2)
+
+CHROOT_DIR = TEMP_DIR
+
+
+if not os.path.isdir(CHROOT_DIR):
+  print(CHROOT_DIR + " does not exist.")
+  os.mkdir(CHROOT_DIR)
+  # TODO: Check creation
+else:
+  if not args.auto_yes
+    print(CHROOT_DIR + " directory already exists")
+    val = input("Would you like to continue chroot setup? [y/N]")
```

Comment:
We could probably do something like this to make this more DRY:

```python

def input_yes(*args, **kwargs):
    return 'y'

if args.auto_yes:
    input = input_yes
```

Reply:
Fixed up in d80220a